src/libmongoc/src/mongoc/mongoc-cursor.c

```diff
@@ -1005,6 +1010,12 @@ _mongoc_cursor_run_command (mongoc_cursor_t *cursor,
       GOTO (done);
    }
 
+   /* Exhaust cursors with OP_MSG not yet supported. Fallback to normal cursor.
+    */
+   if (parts.user_query_flags & MONGOC_QUERY_EXHAUST) {
+      parts.user_query_flags ^= MONGOC_QUERY_EXHAUST;
+   }
```

Comment:
Correct, this was primarily to ensure the state of the query flags reflected actual behavior (exhaust option being ignored), since if OP_MSG is eventually supported, it is likely through the query flags that the option will be communicated to lower-level abstractions. Will document this context with a comment.